src/linker/utilities/spark_utils.py

```diff
@@ -3,24 +3,24 @@
 import tempfile
 from pathlib import Path
 from time import sleep
-from typing import List, TextIO
+from typing import Any, Dict, List, Optional, TextIO
 
 from loguru import logger
-from numpy import diag
 
-from linker.configuration import Config
 from linker.utilities.paths import CONTAINER_DIR
 from linker.utilities.slurm_utils import get_slurm_drmaa, submit_spark_cluster_job
 
 
 def build_cluster(
-    config: Config,
+    session: Optional["drmaa.Session"],
+    environment: Dict[str, Any],
+    step_id: str,
     results_dir: Path,
     diagnostics_dir: Path,
     input_data: List[Path],
     preserve_logs: bool = False,
```

Comment:
It occurs to me that by deleting the linker build-spark-cluster command we've lost any way of preserving the logs. Is this a feature we'd like to support? If so, should it be a command-line option (I don't think so) or should it be a spark comfiguration key (my preference)? @zmbc?

Reply:
I think a spark configuration key makes sense. And personally, I'd make the default True.

Reply:
As discussed I've removed the option and now we always save the files (but in the diagnostics directory instead of working directory)